kernel/signal.c

```diff
@@ -286,6 +286,16 @@ static long _handle_one_signal(
 
         if ((action->flags & SA_SIGINFO) != 0)
         {
+            // pass stack limits if using alternate stack.
+            // dotnet runtime uses the uc_stack field to detect if its running
+            // on an alternate stack.
+            if (use_alt_stack)
+            {
+                context.uc_stack.ss_sp = altstack->ss_sp;
+                context.uc_stack.ss_size = altstack->ss_size;
+                context.uc_stack.ss_flags = altstack->ss_flags;
```

Comment:
`ss_flags.SS_ONSTACK` is set on line 317. Probably should move the code block down there.

Reply:
Thanks for catching that. Fixed.